backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java

```diff
@@ -202,32 +202,54 @@ private void checkIfFileIsAlreadyInsideRoot(FileMetadataEntity fileMetadataEntit
     }
   }
 
-  private FileMetadataEntity findFileMetaDataEntity(UUID fileId) {
-    return fileMetadataRepository.findById(fileId).orElseThrow(FileNotFoundException::new);
-  }
-
   private void saveMoveFile(FileMetadataEntity fileMetadataEntity) {
     fileMetadataRepository.save(fileMetadataEntity);
     actionService.recordAction(Collections.singleton(fileMetadataEntity), ActionType.MOVE);
   }
 
-  private List<FileResponse> asFileResponseList(List<FileMetadataEntity> allFiles) {
-    return allFiles.stream()
-        .map(folder -> modelMapper.map(folder, FileResponse.class))
-        .collect(Collectors.toList());
-  }
-
   @Override
   @RecordAction(objectId = "[0].toString()", actionType = ActionType.DELETE)
   public void deleteFile(UUID fileId) {
-    FileMetadataEntity fileMetadataEntity = getFileMetadataEntity(fileId);
+    FileMetadataEntity fileMetadataEntity = findFileMetadataById(fileId);
     if (fileMetadataEntity.isDeleted()) {
       throw new FileAlreadyDeletedException();
     }
     fileMetadataEntity.delete();
     fileMetadataRepository.save(fileMetadataEntity);
   }
 
+  private FileMetadataEntity saveContentAndMetadata(
+      FileMetadataEntity fileMetadata, MultipartFile multipartFile) {
+
+    checkOnlyCreatorsCanAccessPrivateBuckets(fileMetadata);
```

Comment:
Move these checks in some new service, bucket validation or access service

Reply:
there are quite a few of these checks in this service, perhaps as new PR?